### ISSUE TITLE
Confusing message and structure field tags

### ISSUE THREAD

**Issue by kerhong** [gtker/wow_messages#91]
Currently these tags are being used on fields of messages and structures, looks like they could be simplified and deduplicated.

Perhaps also a list of valid tags could be added and validated against?

- `maximum_length`
    - used only in `SMSG_PARTY_COMMAND_RESULT`
    - used on `CString`
    - I assume it's `str.len() <= tag`
    - :question: doesn't seem to be used in codegen (either validation of the field in particular or the read `contains(&body_size)` check)
- `max_length`
    - used only in `SMSG_GMRESPONSE_RECEIVED`
    - used on `Cstring` and `CString[4]`
    - I assume it's `str.len() <= tag`
    - :question: duplicate of `maximum_length`?
    - :question: doesn't seem to be used in codegen (either validation of the field in particular or the read `contains(&body_size)` check)
- `maximum_value`
    - used only in `CMSG_WHO`
    - used on `u32`
    - I assume it's `val <= tag`
    - :question: doesn't seem to be used in codegen
- `maximum_valid_value`
    - used only in `SMSG_QUESTUPDATE_ADD_ITEM`
    - used on `u32`
    - I assume it's `value <= tag`
    - :question: duplicate of `maximum_value`?
    - :question: doesn't seem to be used in codegen
- `minimum_valid_value`
    - used only in `SMSG_QUESTUPDATE_ADD_ITEM`
    - used on `u32`
    - I assume it's `value >= tag`
    - :question: doesn't seem to be used in codegen
- `minimum_legal_value`
    - used only in `CMSG_SPLIT_ITEM`
    - used on `u32`
    - I assume it's `val >= tag`
    - :question: duplicate of `minimum_valid_value`?
    - :question: doesn't seem to be used in codegen
- `valid_range`
    - used for several messages
    - used on `u8`, `u32`
    - I assume it's `value >= tag[0] && value <= tag[1]`
    - :question: overlaps with the min/max value tags?
    - :question: doesn't seem to be used in codegen
- `compressed`
    - used for several messages
    - used on `u8[-]`
    - :question: should be used (but isn't) on `CMD_SURVEY_RESULT.data`?

**Comment by gtker** [comment]
Sorry for not responding, I remember reading this post but then I forgot to reply to it.

None of them are used in codegen right now. It was planned but isn't really high on the priority list since so many other things need attention instead.

I'll deduplicate the tags and make them show up in the IR so that people can actually use them, then add a check for unknown tags.

**Comment by gtker** [comment]
I have deduplicated the unused tags and passed them through to the IR.

The reason why `CMD_SURVEY_RESULT.data` doesn't have the `compressed` tag is it doesn't follow the same rules as the compressed world data that is prepended with a `u32` containing the decompressed length. Instead it has a member with the compressed length.

I'm also not sure of the exact compression method used.

Everything in this issue has been fixed except for the field not being marked compressed.
